src/main/java/io/cdap/plugin/gcp/bigquery/relational/BigQueryRelation.java

```diff
@@ -349,6 +349,16 @@ public Relation window(WindowAggregationDefinition definition) {
     return new BigQueryRelation(datasetName, columns, featureFlagsProvider, this, supplier);
   }
 
+  /**
+   * Adds schema information to the relation. This can be used for validation purposes.
+   *
+   * @param schema The schema.
+   * @return A new relation with the schema added.
+   */
+  public Relation addSchema(@Nullable Schema schema) {
```

Comment:
Please validate there is no existing schema unless you expect schema overrides

Reply:
I don't know how do you plan to use this method. Generally it's better to introduce methods when they are needed (even with mock implementation) than visa versa.

Reply:
Whangler should have nothing to do with schema management. How would Wrangler know the schema of a SQL expression?

Reply:
Also it's unneded burden on the plugin. Given known schema of the original relation and known schema of all expressions, resulting schema should be automatically contructed by the platform